mcda/utils/utils_for_main.py

```diff
@@ -452,18 +498,26 @@ def save_dict(dictionary: dict, folder_path: str, filename: str):
     :param filename: str
     :return: None
     """
+    custom_output_path = os.environ.get('PROMCDA_OUTPUT_PATH')  # check if an environmental variable is set
+    output_directory_path = custom_output_path if custom_output_path else DEFAULT_OUTPUT_DIRECTORY_PATH
+
     timestamp = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
     new_filename = f"{timestamp}_{filename}"
-    result_path = os.path.join(folder_path, new_filename)
 
-    with open(result_path, 'wb') as fp:
+    full_output_path = os.path.join(output_directory_path, folder_path, new_filename)
```

Comment:
I would propose to extract 4 lines of common code that checks and ensures output path into some method.